hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/KeyProvider.java

```diff
@@ -639,13 +639,16 @@ public void invalidateCache(String name) throws IOException {
   public abstract void flush() throws IOException;
 
   /**
-   * Split the versionName in to a base name. Converts "/aaa/bbb/3" to
+   * Split the versionName in to a base name. Converts "/aaa/bbb@3" to
    * "/aaa/bbb".
    * @param versionName the version name to split
    * @return the base name of the key
    * @throws IOException raised on errors performing I/O.
    */
   public static String getBaseName(String versionName) throws IOException {
+    if (versionName == null) {
+      throw new IOException("Null string found in key path");
```

Comment:
it's ok to raise NPEs on bad data, so we can just use Objects.nonNull() but at least add a meaningful message

Reply:
This has been addressed, please review.